ucp/_libs/core.pyx

```diff
@@ -289,6 +293,13 @@ cdef class ApplicationContext:
         return self.config
 
 
+cdef struct Tags:
+    uint64_t send_tag
+    uint64_t recv_tag
+    uint64_t ctrl_send_tag
+    uint64_t ctrl_recv_tag
```

Comment:
Why are there separate send and receive tags?

Reply:
Am mirroring [this code]( https://github.com/rapidsai/ucx-py/blob/7b739a48dc4b647edd7234424e2dd94720606c86/ucp/_libs/core.pyx#L329-L332 ).

Reply:
Perhaps @madsbk can clarify on why he structured it that way? 🙂

Reply:
@MattBBaker, I don't think that we need separate send and receive tags anymore but they are nice to have in the log messages (making it easy to follow the communication flow).
If we use the pointer address of Endpoints as IDs in the log instead, which is the same UCX does, we can combine the send/recv tags and still have a nice log.

But I think we should merge this PR and make another PR that combines the send/recv tags.

Reply:
It won't help with the tags, because I need those to be unified to avoid the deadlock. But I should chime in on the API issues.

Reply:
@mt-jones check https://github.com/rapidsai/ucx-py/pull/238